### PR TITLE
Added a directory search for SteamDeck

### PR DESCRIPTION
**Pull request by ianrice07**

### CODE REVIEW

File: Runtime/Native/Windows/NativeClient.cs [hunk 1]
```diff
@@ -250,13 +250,24 @@ public void SetAttribute(string key, string value)
         public IEnumerator SendMinidumpOnStartup(ICollection<string> clientAttachments, IBacktraceApi backtraceApi)
         {
             // Path to the native crash directory
-            string nativeCrashesDir = Path.Combine(
-                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
-                    string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName));
+            string tmpDir = string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName);
+            string sdeckCrashPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), tmpDir);
+            string windowsCrashPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), tmpDir);
```
Comment (konraddysput) [inline Runtime/Native/Windows/NativeClient.cs:255]:
```suggestion
            string tempDirectory = string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName);
            var crashDirectories = new string[2] { Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), tempDirectory), 
            Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), tempDirectory)
            };
            
            stirng nativeCrashesDir = "";
            for (var directory in crashDirectories) {
                 if (!Directory.Exists) {
                     continue;
                 }
                 nativeCrashesDir = directory;
                 break;
             }
             if (string.IsNullOrEmpty(nativeCrashesDir)) {
                 yield break;
             }
```

File: Runtime/Native/Windows/NativeClient.cs [hunk 2]
```diff
@@ -250,13 +250,24 @@ public void SetAttribute(string key, string value)
         public IEnumerator SendMinidumpOnStartup(ICollection<string> clientAttachments, IBacktraceApi backtraceApi)
         {
             // Path to the native crash directory
-            string nativeCrashesDir = Path.Combine(
-                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
-                    string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName));
+            string tmpDir = string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName);
```
Comment (ianrice07) [inline Runtime/Native/Windows/NativeClient.cs:253]:
```suggestion
            string tempDirectory = string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName);
            string[] crashDirectories = new string[2] {
                Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), tempDirectory),
                Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), tempDirectory)
            };
              
            List<string> nativeCrashesDirs = new List<string>();
            foreach (string direcotry in crashDirectories)
            {
                if (!Directory.Exists(direcotry))
                {
                    continue;
                }
                nativeCrashesDirs.Add(direcotry);
            }

            if (nativeCrashesDirs.Count == 0)
            {
                yield break;
            }

            IDictionary<string, string> attributes = GetScopedAttributes();

            // be sure that error.type attribute provided by default by our library
            // is always present in native attributes.
            attributes[ErrorTypeAttribute] = CrashType;

            foreach (var nativeCrashesDir in nativeCrashesDirs)
            {

                var attachments = clientAttachments == null
                    ? new List<string>()
                    : new List<string>(clientAttachments);

                var crashDirs = Directory.GetDirectories(nativeCrashesDir);

                foreach (var crashDir in crashDirs)
                {
                    var crashDirFullPath = Path.Combine(nativeCrashesDir, crashDir);
                    var crashFiles = Directory.GetFiles(crashDirFullPath);

                    var alreadyUploaded = crashFiles.Any(n => n.EndsWith("backtrace.json"));
                    if (alreadyUploaded)
                    {
                        continue;
                    }
                    var minidumpPath = crashFiles.FirstOrDefault(n => n.EndsWith("crash.dmp"));
                    if (string.IsNullOrEmpty(minidumpPath))
                    {
                        continue;
                    }
                    var dumpAttachment = crashFiles.Concat(attachments).Where(n => n != minidumpPath).ToList();
                    yield return backtraceApi.SendMinidump(minidumpPath, dumpAttachment, attributes, (BacktraceResult result) =>
                    {
                        if (result != null && result.Status == BacktraceResultStatus.Ok)
                        {
                            File.Create(Path.Combine(crashDirFullPath, "backtrace.json"));
                        }
                    });
                }
            }
```

File: Runtime/Native/Windows/NativeClient.cs [hunk 3]
```diff
@@ -249,50 +249,67 @@ public void SetAttribute(string key, string value)
...
-                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
-                    string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName));
+            // Path to the native crash directories
+            string tempDirectory = string.Format("Temp/{0}/{1}/crashes", Application.companyName, Application.productName);
+            string[] crashDirectories = new string[2] {
+                Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.UserProfile), tempDirectory),
+                Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), tempDirectory)
+            };
+              
+            List<string> nativeCrashesDirs = new List<string>();
```
Comment (konraddysput) [inline Runtime/Native/Windows/NativeClient.cs:259]:
```suggestion
            var nativeCrashesDirs = new List<string>();
```

You can use var here

File: Runtime/Native/Windows/NativeClient.cs [hunk 4]
```diff
@@ -249,50 +249,67 @@ public void SetAttribute(string key, string value)
...
                 yield break;
             }
 
-            var attachments = clientAttachments == null
-                ? new List<string>()
-                : new List<string>(clientAttachments);
-
-            var crashDirs = Directory.GetDirectories(nativeCrashesDir);
-
             IDictionary<string, string> attributes = GetScopedAttributes();
```
Comment (konraddysput) [inline Runtime/Native/Windows/NativeClient.cs:274]:
```suggestion
            var attributes = GetScopedAttributes();
```
you can use var here. You should use var whenever it's possible and when you don't need to explicitly say what is the type of something

**Review by konraddysput — APPROVED** [review]

File: Runtime/Native/Windows/NativeClient.cs (hunk 1 above)
Comment (ianrice07) [reply Runtime/Native/Windows/NativeClient.cs]: This doesn't fully resolve an issue I noticed where the Deck will sometimes use the normal Windows path as opposed to the one that we found it uses.  Seems to crop up randomly.

File: Runtime/Native/Windows/NativeClient.cs (hunk 2 above)
Comment (ianrice07) [reply Runtime/Native/Windows/NativeClient.cs]: Changes commited.

File: Runtime/Native/Windows/NativeClient.cs (hunk 4 above)
Comment (ianrice07) [reply Runtime/Native/Windows/NativeClient.cs]: Changes applied.

**Pull request merged**